### ISSUE TITLE
Functions with dependencies are not updated

### ISSUE THREAD

**Issue by phoet** [aidenybai/lucia#17]
**Describe the bug**
When rendering content (ie text) my expectation was that i could put the expression into a function on the context object and have the content updated whenever the depencency (ie `count`) is changed.

**To Reproduce**
Intuitively I would expect both buttons to show the same text:

```
<div l-use="{ count: 0, text() { return this.count + ' hi'; } }">
  <button l-text="text()" l-on:click="++count">0</button>
  <button l-text="count + ' hi'" l-on:click="++count">0</button>
</div>
```

If this is not intended or should be done any other way, it's probably because the documentation is a bit scarce...

**Comment by shadowtime2000** [comment]
@phoet I don't believe it is actually possible with our current structure to update the dependencies of functions. Use pure functions like this:
```html
<div l-use="{ count: 0, text(count) { return count + ' hi'; } }">
  <button l-text="text(count)" l-on:click="++count">0</button>
  <button l-text="count + ' hi'" l-on:click="++count">0</button>
</div>
```

**Comment by aidenybai** [comment]
This is possible if you have an external function that returns state

**Comment by phoet** [comment]
Fair enough. 

Since the documentation has references to Vue, I was expecting some kind of similar functionality when it comes to methods used in bindings.

There should at least be a minimal example in the documentation that shows the inteded use.

**Comment by shadowtime2000** [comment]
Good point, the documentation doesn't reflect the intended use. I will update that. @phoet feel free to open a PR to update it yourself too.

**Comment by phoet** [comment]
I find it especially irritating, as the `l-on:click` handlers can be functions and update the state. so from my point, following the principle of least surprise, binding a method to `l-text` should always return an updated value as well.

**Comment by aidenybai** [comment]
Issue seems to be here specifically, need to somehow be able to compare `value` and `view[key].toString()` and see that it is accessing affected state. Although this is a temporary solution, we're currently working on a full vdom refactor under the `vdom-refactor` branch.

![image](https://user-images.githubusercontent.com/38025074/97812743-49fd1080-1c38-11eb-8d62-eb98aa1fc40b.png)

**Comment by aidenybai** [comment]
Fixed in ca44c2e953cce44fce28f9debf3db2bb0b88c18f
